### ISSUE TITLE
Create Spaces [16]

### ISSUE THREAD

**Issue by michaelkleinhenz** [fabric8io/fabric8-planner#667]
# Description

As a user, I want to create projects from the system. A project is a higher context for work items and other data in the system. A work item always "belongs" to exactly one project and all data in the system is always displayed in the context of a specific project. Projects are "owners" of work items and work item links. 
# Functional Acceptance Criteria
1. The user can create a project from the "create project" view.
2. Creating a project requires a name (alphanumeric value).
3. It is not possible to create a project with an empty name.
4. Projects are working in a way that users in the system are global and can be associated with more than one project (will be a later story).
5. Projects are working in a way that cross-linking of work items between projects is possible (will be a later story).

**Note:** as mentioned above, users are "global" to the system and can be "member" of more that one project at the same time. This is not a direct functional requirement, but a hint for design decisions. 

**Note:** projects are "owners" of work items and work item links. Work items and projects should be designed that cross-linking of work items between projects are possible later. TBD is the ownership of the work item link instance on cross-linked work items. This is not a direct functional requirement, but a hint for design decisions. 

**Note:** on the scope of this story: as part this story, we will extend our object model by introducing a 1:n relationship between the newly created "project" and work items. On the back end, we will have to introduce a CRU(D) service for projects and extend work item creation with a mandatory "project" parameter. How that parameter is passed and stored needs to be discussed as part of implementation.
The front end needs to add UI to create a new project and introduce the notion of a "selected" project that is passed when creating a work item. As part of migration, we will have to create a default project and assign all existing issues to that project.
_**Out of scope** (explicitly not addressed by this user story): authorization, orgs, teams, areas, configuration of process for a project (WI types, etc.) and moving work items between projects._

# Non-functional Acceptance Criteria
1. The "create project" view is easily accessible from anywhere in the system UI.
2. Creating a project adds the new project in-place to the project selector.
3. The result of the implementation of this story should also be a basic data model and a concept for organizing projects and other entities.

# Dependent Stories
1. Userstory #358

# Tasks

* Create Repository for project crud(l) #509
* Create API Endpoint for project crudl #510
* Update work item create/update to include optional "project field" if empty, assign to a default project
  *migrate all existing work items to default project #511
* Update search services (wi list and @searchrepo) to allow filtering by project #512
* Make project field in work item create/update mandatory
  * this can be done once the front end is ready to pass a project with work item create/update (#358)

**Comment by michaelkleinhenz** [comment]
Not in Planner scope.
